### ISSUE TITLE
Ref not updating when conditionally applied

### ISSUE THREAD

**Issue by stramel** [wellyshen/react-cool-inview#493]
# Bug Report

## How to Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Conditionally apply the `ref`
2. Scroll to bottom
3. Observe no callbacks fired and no entry exists

## CodeSandbox Link

https://codesandbox.io/s/react-cool-inviewissues493-yks98

## Expected Behavior

I would expect the observer to fire callbacks

## Screenshots

![image](https://user-images.githubusercontent.com/855184/113337583-95057780-92ed-11eb-9667-28b5d74c34de.png)

## Your Environment

- Device: MacBook Pro
- OS: macOS X
- Browser: Chrome
- Version: 1.2.2

## Additional Information

Seems to be an issue when `ref` is conditionally set.

```jsx
{hasNextPage && <div ref={triggerRef}>Loading...</div>}
```

```jsx
{items.map((item, i) => {
	  return i === items.length - 1 ? (
	    <div ref={lastItemRef} key={item.id}>
	      {item.id}
	    </div>
	  ) : (
	    <div key={item.id}>{item.id}</div>
	  );
})}
```

**Comment by stramel** [comment]
`RefCallback` may help here. Possibly related conversation: https://github.com/popperjs/react-popper/issues/241#issuecomment-591411605

**Comment by wellyshen** [comment]
@stramel See if the [latest version](https://github.com/wellyshen/react-cool-inview/releases/tag/v1.3.0) helps or not.

**Comment by stramel** [comment]
@wellyshen Thank you for a quick patch! Unfortunately, I'm unable to get it to work until after the page has performed another render where the element is visible. (Normally after HMR it works) Initial loads still don't work.

Option 1 from the section referenced: https://codesandbox.io/s/winter-cloud-b52j4?file=/src/App.js
![image](https://user-images.githubusercontent.com/855184/113449656-eaac5380-93c3-11eb-84cc-01c3802d7545.png)

Here is the initial example modified with the `observe` solution: https://codesandbox.io/s/funny-dawn-q8mlw?file=/src/index.js

**Comment by wellyshen** [comment]
@stramel Sorry my fault, will fix it by following your use case in recent days.

**Comment by stramel** [comment]
@wellyshen No worries! I appreciate your time on these bugs. It will be very beneficial as we start to use this in our projects at my company!

**Comment by wellyshen** [comment]
@stramel Thank you for your support, I think this issue should be fixed in [v1.3.1](https://github.com/wellyshen/react-cool-inview/releases/tag/v1.3.1). BTW, I found you might forget to de-structure the new created items from the demo code.

```diff
async onEnter({ observe, unobserve }) {
    unobserve();

    await new Promise((res) => {
      // Simulating data coming in from API
      setTimeout(() => {
        -setItems([...items, createItems()]);
       +setItems([...items, ...createItems()]);
        res();
      }, 200);
    });
    observe();
  }
```

**Comment by stramel** [comment]
@wellyshen Ah! Thank you for catching that typo! I checked with version 2 and it worked very well! Thank you for the fixes!
